### DISCUSSION TITLE
Votifier - Add option "Only run commands when player is online"

### DISCUSSION THREAD

**Discussion by Nesseley** [Zrips/CMI#6997] [Enhancement]
Hello!

Would love to see a feature where there is an option to not perform the CommandOnVote commands until the player is online on the server.

I give out keys for opening chests, and I use CMI kit to give the keys on vote. Problem is that the keys drop onto the floor if the inventory is full, where they might despawn if someone else is in the area.

Hope to see this implemented in the future.

**Comment by mrfloris** [comment]
I don't know if this has already been implemented; I don't remember reading it in the changelog. IF NOT: I would love to see this too yes. 

I know we can use ifonline! specialized command, but I don't know how to 'queue' the rewards, so they get stacked up and pay out when they are online (with say a 14 days max, we're not going to reward for not playing).
Ideally

**Comment by mrfloris** [comment]
I don't know if this has already been implemented; I don't remember reading it in the changelog. IF NOT: I would love to see this too yes. 

I know we can use ifonline! specialized command, but I don't know how to 'queue' the rewards, so they get stacked up and pay out when they are online (with say a 14 days max, we're not going to reward for not playing).

Ideally, id' like to say if they join and go to THIS world, give THAT reward, and if they join THEOTHER world, give THEOTHER reward, so we can tell survival players to get kit5 and skyblock players kit6.

**Comment by Nesseley** [comment]
A queue with a limit would be the best yeah, as far as I know this has not been implemented yet :)

**Comment by CloudeLecaw** [comment]
reward section - command useage- ifonline

```
CommandsOnVote:
  - cmi kit Votis [playerName] -s
  - cmi ifonline [playerName] aach add 1 Custom.votes [playerName]
  - cmi ifoffline [playerName] cmi give [playerName] diamond 1
 ```

a bad idea with the quene.. specialy when you have more sites. on our server we registered 7 vote sites so this would mean constant check of 7 quenes per player if all are voting offlne..

so an idea here might be to use commands that work too while player is offline like the money command. or teach player to not log off with full inventory if they wanne vote. well or useage of meta and specialized commands

**Reply by mrfloris** [reply]
What if the offline player has a full inventory, when they log on, are the items on the floor? If so, what if the chunk is loaded somehow because others are nearby and they a- either pick it up, or b- these items despawn? 
This is just one concern, I don't think this is a viable solution.

**Comment by CloudeLecaw** [comment]
so create the meta as vote reward (if player if offline) else give him the reward (if inventory is not full else create meta)


on logon do the meta check + if inventory is full
give reward and remove meta if condition is true.

for more condition take a look at specialized commands.
https://www.zrips.net/cmi/commands/specialized/

About timed despawn .. not the first time auto despawn plugins like NoLag and co. or a cmi scheduler for groundclean avoid people that droped items or got items droped at the wrong time.
